### PR TITLE
[feat][doc] PIP-264: Add OpenTelemetry producer metrics reference

### PR DESCRIPTION
**Pull request by dragosvictor**
### ✅ Contribution Checklist

- [x] I read the [contribution guide](https://pulsar.apache.org/contribute/document-contribution/)
- [x] I updated the [versioned docs](https://pulsar.apache.org/contribute/document-contribution/#update-versioned-docs)

### CODE REVIEW

File: docs/reference-metrics-opentelemetry.md [hunk 1]
```diff
@@ -402,3 +402,62 @@ The number of pending topic load operations in the broker. When it reaches thres
...
+  * `pulsar.producer.access_mode` - The access mode of the producer. Can be one of:
+    * `shared`
+    * `exclusive`
+    * `wait_for_exclusive`
+    * `exclusive_with_fencing`
+
+#### pulsar.broker.producer.message.incoming.size
+The total number of messages bytes received from this producer.
+* Type: Counter
+* Unit: `By`
```
Comment (heesung-sn) [inline docs/reference-metrics-opentelemetry.md:430]: Is `By` bytes?

File: docs/reference-metrics-opentelemetry.md [hunk 2]
```diff
@@ -402,3 +402,62 @@ The number of pending topic load operations in the broker. When it reaches thres
...
 * Type: UpDownCounter
 * Unit: `{operation}`
+
+### Producer metrics
+
+#### pulsar.broker.producer.message.incoming.count
+The total number of messages received from this producer.
+* Type: Counter
+* Unit: `{message}`
+* Attributes:
```
Comment (heesung-sn) [inline docs/reference-metrics-opentelemetry.md:412]: Do we include the `pulsar.cluster` info too?

File: docs/reference-metrics-opentelemetry.md [hunk 3]
```diff
@@ -402,3 +402,62 @@ The number of pending topic load operations in the broker. When it reaches thres
...
+  * `pulsar.producer.name` - The name of the producer.
+  * `pulsar.producer.id` - The ID of the producer.
+  * `pulsar.producer.access_mode` - The access mode of the producer. Can be one of:
+    * `shared`
+    * `exclusive`
+    * `wait_for_exclusive`
+    * `exclusive_with_fencing`
+
+#### pulsar.broker.producer.message.drop.count
+The total number of messages dropped from this producer.
```
Comment (heesung-sn) [inline docs/reference-metrics-opentelemetry.md:447]: could we clarify "dropped messages" more or put a reference link?

**Review by heesung-sn — APPROVED** [review]

File: docs/reference-metrics-opentelemetry.md (hunk 1 above)
Comment (dragosvictor) [reply docs/reference-metrics-opentelemetry.md:430]: Yes. OpenTelemetry mandates using international units where possible. For bytes, the unit is then `By`. See https://opentelemetry.io/docs/specs/semconv/general/metrics/#instrument-units and https://unitsofmeasure.org/ucum#para-48.

File: docs/reference-metrics-opentelemetry.md (hunk 3 above)
Comment (dragosvictor) [reply docs/reference-metrics-opentelemetry.md:447]: How does this sound?

```suggestion
The total number of messages dropped from this producer. Pending messages published to non-persistent topics can be dropped when they accumulate in excess of broker config option `maxConcurrentNonPersistentMessagePerConnection`.
```

**Pull request merged**